host/tests/dsClasses/dsHost.py

```diff
@@ -0,0 +1,161 @@
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import yaml
+import os
+import sys
+import re
+from enum import Enum, auto
+
+# Add parent outside of the class directory
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(dir_path+"/../")
+
+from raft.framework.plugins.ut_raft.configRead import ConfigRead
+from raft.framework.plugins.ut_raft.utSuiteNavigator import UTSuiteNavigatorClass
+from raft.framework.plugins.ut_raft.interactiveShell import InteractiveShell
+
+class dsHostClass():
+
+    moduleName = "dsHost"
+    menuConfig =  dir_path + "/dsHost_test_suite.yml"
+    testSuite = "L3 dsHost"
+
+    """
+    Device Settings Host Class
+
+    This module provides common extensions for device Settings Host Module.
+    """
+    def __init__(self, deviceProfilePath:str, session=None ):
+        """
+        Initializes the dsHost class function.
+        """
+        self.deviceProfile = ConfigRead( deviceProfilePath, self.moduleName)
+        self.utMenu        = UTSuiteNavigatorClass(self.menuConfig, self.moduleName, session)
+
+        self.utMenu.start()
+
+    def searchPattern(self, haystack, pattern):
+        match = re.search(pattern, haystack)
+        if match:
+            return match.group(1)
+        return None
+
+    def initialise(self):
+        """
+        Initializes the device settings Host module.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select( self.testSuite, "Initialize dsHost")
+    
+    def getCPUTemperature(self):
+        """
+        Get the CPU temperature.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select( self.testSuite, "Get CPU Temperature")
+        typeStatusPattern = r'Result dsGetCPUTemperature\(cpuTemperature: (?P<cpuTemperature>[-+]?\d*\.\d+|\d+)\)'
+        match = self.searchPattern(result, typeStatusPattern)
+
+        if match:
+            temperature = float(match)
+            return temperature
+        else:
+            print("Error: Could not get CPU temperature")
+            return None
+
+    def getSoCID(self):
+        """
+        Get the SoC ID.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        result = self.utMenu.select( self.testSuite, "Get SoC ID")
+
+    def getHostEDID(self):
```

Comment:
no need to use it if we are not using it.

Reply:
Removed, as discussed.